src/features/projects/ProjectRegistForm.tsx

```diff
@@ -0,0 +1,46 @@
+import { createProject } from 'api';
+import React, { useCallback, useRef } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+// DocumentList represents the list of documents in the application.
+export function ProjectRegistForm() {
+  const navigate = useNavigate();
+  const nameRef = useRef<HTMLInputElement | null>(null);
+  const handleSumbit = useCallback(
+    async (e) => {
+      e.preventDefault();
```

Comment:
What issue does this code solve? It would be nice to add comments if necessary.

Reply:
I added `e.preventDefault()` to prevent refreshing the page immediately on submit.